### PR TITLE
Animations: Refactor the code that creates additive animations

### PR DESCRIPTION
**Pull request by Popov72**
* added some options to drive more precisely the creation of additive animations
* added a helper function to clip an animation group given a range of frames
* fixed the `clone` / `serialize` / `parse` methods of the `AnimationGroup` class

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14278/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14278/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14278/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14278/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14278/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14278/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14278/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14278/merge#BCU1XR#0

File: packages/dev/core/src/Animations/animationGroup.ts [hunk 1]
```diff
@@ -914,68 +974,171 @@ export class AnimationGroup implements IDisposable {
...
+     * @param fromFrame defines the lower bound of the range
+     * @param toFrame defines the upper bound of the range
+     * @param name defines the name of the new animation group. If not provided, use the same name as animationGroup
+     * @param dontCloneAnimations defines whether or not the animations should be cloned before clipping the frames. Default is false, so animations will be cloned
+     * @returns a new animation group stripped from all the frames outside the given range
+     */
+    public static ClipFrames(sourceAnimationGroup: AnimationGroup, fromFrame: number, toFrame: number, name?: string, dontCloneAnimations?: boolean): AnimationGroup {
+        const animationGroup = sourceAnimationGroup.clone(name || sourceAnimationGroup.name);
+
+        return this.ClipFramesInPlace(animationGroup, fromFrame, toFrame, dontCloneAnimations);
```
Comment (RaananW) [inline packages/dev/core/src/Animations/animationGroup.ts:1093]: super-nit

```suggestion
        return AnimationGroup.ClipFramesInPlace(animationGroup, fromFrame, toFrame, dontCloneAnimations);
```

File: packages/dev/core/src/Animations/animationGroup.ts [hunk 2]
```diff
@@ -914,68 +974,171 @@ export class AnimationGroup implements IDisposable {
...
      * @param name defines the name of the new animation group. If not provided, use the same name as animationGroup
+     * @param dontCloneAnimations defines whether or not the animations should be cloned before clipping the keys. Default is false, so animations will be cloned
      * @returns a new animation group stripped from all the keys outside the given range
      */
-    public static ClipKeys(sourceAnimationGroup: AnimationGroup, fromKey: number, toKey: number, name?: string): AnimationGroup {
+    public static ClipKeys(sourceAnimationGroup: AnimationGroup, fromKey: number, toKey: number, name?: string, dontCloneAnimations?: boolean): AnimationGroup {
         const animationGroup = sourceAnimationGroup.clone(name || sourceAnimationGroup.name);
 
-        return this.ClipKeysInPlace(animationGroup, fromKey, toKey);
+        return this.ClipKeysInPlace(animationGroup, fromKey, toKey, dontCloneAnimations);
```
Comment (RaananW) [inline packages/dev/core/src/Animations/animationGroup.ts:1066]:
```suggestion
        return AnimationGroup.ClipKeysInPlace(animationGroup, fromKey, toKey, dontCloneAnimations);
```

**Review by deltakosh — APPROVED** [review]

**Pull request merged**